File: 78-subsets/78-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:

        result = []
        
        
        def backtrack(i,curr=[]):
            
            if i == len(nums):
                result.append(curr[::])
                return
            
            curr.append(nums[i])
            backtrack(i+1,curr)
            curr.pop()
            backtrack(i+1,curr)
            
        backtrack(0)
        
        return result

File: 78-subsets/78-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    
    result = []
    
    const backtrack = (i,curr=[])=>{
        
        if(i==nums.length){
            result.push([...curr])
            return
        }
        
        curr.push(nums[i])
        backtrack(i+1,curr)
        curr.pop()
        backtrack(i+1,curr)
    }
    
    backtrack(0,[])
    
    return result
    
    
};